app/frontend/components/pages/txHistory/transactionHistory.tsx

```diff
@@ -74,52 +75,93 @@ interface Props {
   transactionHistory: any
 }
 
-const ExportCSV = ({transactionHistory}) => {
-  const formatTransactionDate = (ctbTimeIssued) =>
-    moment.utc(new Date(ctbTimeIssued * 1000)).format('MM/DD/YYYY hh:mm A [UTC]')
+const ExportCSV = ({transactionHistory, stakingHistory}) => {
+  const withdrawalHistoryTxHashes = stakingHistory
+    .filter((item) => item.type === StakingHistoryItemType.RewardWithdrawal)
+    .map((withdrawal) => withdrawal.txHash)
+
+  const stakingRewards = stakingHistory.filter(
+    (item) => item.type === StakingHistoryItemType.StakingReward
+  )
 
   const delimiter = ','
   const rowsDelimiter = '\n'
 
-  const headers =
-    `Type${delimiter}` +
-    `Received amount${delimiter}` +
-    `Received currency${delimiter}` +
-    `Sent amount${delimiter}` +
-    `Sent currency${delimiter}` +
-    `Fee amount${delimiter}` +
-    `Fee currency${delimiter}` +
-    `Transaction ID${delimiter}` +
-    `Date${delimiter}`
-
-  const rows = transactionHistory.map((transaction) => {
-    if (transaction.effect > 0) {
-      return (
-        `${'Received'}${delimiter}` +
-        `${printAda(transaction.effect as Lovelace)}${delimiter}` +
-        `ADA${delimiter}` +
-        `${delimiter}` +
-        `${delimiter}` +
-        `${delimiter}` +
-        `${delimiter}` +
-        `${transaction.ctbId}${delimiter}` +
-        `${formatTransactionDate(transaction.ctbTimeIssued)}${delimiter}`
-      )
+  const headers = [
+    'Type',
+    'Received amount',
+    'Received currency',
+    'Sent amount',
+    'Sent currency',
+    'Fee amount',
+    'Fee currency',
+    'Transaction ID',
+    'Date',
+  ]
+    .map((header) => `${header}${delimiter}`)
+    .join('')
+
+  const transactionTypes = {
+    received: 'Received',
+    sent: 'Sent',
+    rewardAwarded: 'Reward awarded',
+    rewardWithdrawal: 'Reward withdrawal',
+  }
+
+  const transactionsEntries = transactionHistory.map((transaction) => {
+    const common = {
+      txHash: transaction.ctbId,
+      dateTime: moment.utc(new Date(transaction.ctbTimeIssued * 1000)),
+    }
+
+    if (withdrawalHistoryTxHashes.includes(transaction.ctbId)) {
```

Comment:
It's not a problem with a small number of withdrawals, but their count can get ramped up over time. I would therefore make withdrawalHistoryTxHashes into a hash table, so we have o(1) lookups here for each transaction

Reply:
Fixed